### ISSUE TITLE
Local storage rotation keeps trying to delete an old backup that doesn't exist anymore

### ISSUE THREAD

**Issue by mtx-z** [backup/backup#948]
## What went wrong?

I have this error message twice a day, every time the backup local storage is running.
The first time I encountered this, I checked the filesystem and there was no folder matching the one the script is trying to delete. 
I deleted all folders from the "15.07.2019" date manually. But everyday, it keeps trying to delete this folder. And so, rotation is not executed: all backup runs successfully and stored. 

It seems (I need to double-check it) that the rotation is working. I never have more than 60 backups stored, which respect my script config (58 actually). But I still received twice a day (backup is twice a day) the warning about this delete.

```
[2019/08/30 12:05:52][warn] Storage::Cycler::Error: There was a problem removing the following package:
[2019/08/30 12:05:52][warn]   Trigger: database :: Dated: 15.07.2019_01h40m18s
[2019/08/30 12:05:52][warn]   Package included the following 1 file(s):
[2019/08/30 12:05:52][warn]   database.tar
[2019/08/30 12:05:52][warn] --- Wrapped Exception ---
[2019/08/30 12:05:52][warn] Errno::ENOENT: No such file or directory @ unlink_internal - /mnt/mydisk/database/database/15.07.2019_01h40m18s
```

## How is your copy of backup configured?

I do a simple mysql backup that works perfectly. Problem is with the rotation. 
```
 compress_with Gzip do |compression|
    compression.level = 8
    compression.rsyncable = true
 end
 
 store_with Local do |local|
  local.path = '/mnt/kauf6-backup-storage-1/database/'
  # Use a number or a Time object to specify how many backups to keep.
  local.keep = 60
end
```
## Tell us about the computer that runs the backup gem

Linux OS. I have a remote linux server that run the script, and connect to my "servers to backup" to back them up. The "production" servers are not running the backup scripts, it's a specific and dedicated server. 

* Operating system: Debian latest
* Tools or services used by backup: MySQL 5.7
* Backup v4.4.1
* Ruby: ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-linux]

## question

Is there any place where Backup keep tracks of the backup folder names to delete/rotate them? 
Is the script using folder timestamps (meta) to select the one to deletes? 

Thanks a lot for this really cool package anyway.

**Comment by mtx-z** [comment]
Now I think I got a related issue. 
I've a keep of 60 backup. But the cycling is always deleting the last one created, and store a new one. 
So what i have now is: 
- 01/09/2019 to 01/11/2019
- Cycling delete the last one 14/11/2019, to create the 15/11/2019
- so I have missing backup from 01/11 to 15/11. 

Again, is there any log of config file that Backup is using to keep tracks of cycling? Or is it using folders name timestamps to check which is the one to delete? 

Thanks again

**Comment by mtx-z** [comment]
It may, in the end, be related to changing the destination folder name, that messes up with the rotation. 
I had implemented what #641 is explaining. I removed this and will see if it happens again.
